File: backend/QuizHubApplication/DTOs/Responses/QuizAttemptDto.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizHubApplication.DTOs.Responses
{
    public record QuizAttemptDto(
        int QuizAttemptId,
        int UserId,
        string Username,
        int QuizId,
        string QuizTitle,
        string StartedAt,
        string? FinisedAt,
        int TimeLimit,
        int? TimeTakenMin,
        int Score
        )
    {
    }
}

File: frontend/QuizHubFrontend/src/app/features/quiz-attempts/model/QuizAttemptDto.ts
export interface QuizAttemptDto {
  quizAttemptId: number;
  userId: number;
  username: string;
  quizId: number;
  quizTitle: string;
  startedAt: string;
  finisedAt?: string | null;
  timeLimit: number;
  timeTakenMin?: number | null;
  score: number;
}
